1_fetch/src/moving_window_functions.R

```diff
@@ -0,0 +1,139 @@
+
+calc_moving_window_metrics<-function(site_num, window_length,increment, min_yrs_in_window,clean_daily_flow,yearType, 
+                                     drainArea_tab, NE_probs, digits = 3 ){
+  ##subset clean_daily_flow, create a vector of starting years for a moving window analysis, and then
+  ##call the calcFDC function for each starting year.
+  data <- clean_daily_flow %>%
+    filter(site_no == site_num)
+  
+  
+  start_yrs<- seq(from = min(data$year_val), 
+                  to= (max(data$year_val) - window_length+1),
+                  by = increment)
+
+  ##remove any starting years which would result in a window with no data (if there are large data gaps). 
+  df_screen<-bind_rows(map(start_yrs,window_screen,data))
+  start_yrs<-start_yrs[start_yrs %in% df_screen$start_yr[which(df_screen$years_in_window > min_yrs_in_window)]==TRUE]
+  
+  
+  map_out<-map(start_yrs,calc_FDC_subset, 
+               window_length, site_num=site_num, 
+               clean_daily_flow=data,
+               yearType = yearType,
+               drainArea_tab=drainArea_tab,
+               NE_probs=NE_probs,digits=3)
+  data_out<-bind_rows(map_out)
+ 
+  return(data_out)
+}
+
+
+plot_trend_summary<-function(moving_window_metrics,outdir){
+  #normalize metrics and remove NAs (when there is only 1 moving window
+  #for a site,the sd will be 0)
```

Comment:
Is it possible to have NAs when `min_windows` > 1?

Reply:
Hmm, probably not unless all the metrics were identical.  I took it out.  I probably had this in there before I decided to make min_windows a thing.